### PR TITLE
fix: Handle erc721 signature prepare non-hex uids

### PR DESCRIPTION
**Pull request by arcoraven**
## PR-Codex overview
This PR introduces tests for the `erc721SignaturePrepare` function and modifies its implementation to handle various UID scenarios, including string and hex formats. It enhances error handling and updates the structure of the response payload.

### Detailed summary
- Added tests for `erc721SignaturePrepare` in `signaturePrepare.test.ts`.
- Implemented handling for different UID formats (hex, string).
- Updated response structure to include `typedDataPayload`.
- Improved error handling for missing recipients.
- Refactored UID parsing logic and mint payload generation.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/server/routes/contract/extensions/erc721/read/signaturePrepare.ts [hunk 1]
```diff
@@ -84,76 +91,74 @@ const responseSchema = Type.Object({
 
 responseSchema.example = {
   result: {
-    result: {
```
Comment (arcoraven) [inline src/server/routes/contract/extensions/erc721/read/signaturePrepare.ts:87]: There was an extra nested `result` in the example.

File: src/server/routes/contract/extensions/erc721/read/signaturePrepare.ts [hunk 2]
```diff
@@ -212,68 +216,99 @@ export async function erc721SignaturePrepare(fastify: FastifyInstance) {
...
+          )
+        ) {
+          const contract = getContract({
+            chain,
+            client: thirdwebClient,
+            address: parsedCurrency,
+          });
+          _decimals = await decimals({ contract });
+        }
+        parsedPrice = BigInt(Number.parseFloat(price) * 10 ** _decimals);
```
Comment (arcoraven) [inline src/server/routes/contract/extensions/erc721/read/signaturePrepare.ts:262]: `price` was previously not accounting for decimals.

File: src/server/routes/contract/extensions/erc721/read/signaturePrepare.ts [hunk 3]
```diff
@@ -212,68 +216,99 @@ export async function erc721SignaturePrepare(fastify: FastifyInstance) {
...
+          if (uid.length !== 64) {
+            throw createCustomError(
+              '"uid" must be a valid 32-byte hex string.',
+              StatusCodes.BAD_REQUEST,
+              "INVALID_UID",
+            );
+          }
+          parsedUid = uid;
+        } else {
+          parsedUid = stringToHex(uid, { size: 32 });
```
Comment (arcoraven) [inline src/server/routes/contract/extensions/erc721/read/signaturePrepare.ts:288]: This is the main fix, hexifying any input string to a 32-byte hex.

File: src/server/routes/contract/extensions/erc721/read/signaturePrepare.ts [hunk 4]
```diff
@@ -300,73 +335,9 @@ export async function erc721SignaturePrepare(fastify: FastifyInstance) {
   });
 }
 
-type GenerateMintSignaturePayloadOptions = Omit<
```
Comment (arcoraven) [inline src/server/routes/contract/extensions/erc721/read/signaturePrepare.ts:303]: Removed this unnecessary abstraction. The main handler function now does all transformations on the input fields.

**Review by d4mr — APPROVED** [review]

**Pull request merged**